src/main/kotlin/skytils/skytilsmod/features/impl/misc/LockOrb.kt

```diff
@@ -37,30 +39,38 @@ class LockOrb {
         val heldOrb = PowerOrbs.getPowerOrbMatchingItemId(itemId) ?: return
         val orbs = mc.theWorld.getEntities(
             EntityArmorStand::class.java
-        ) { entity: EntityArmorStand? ->
-            val orb = PowerOrbs.getPowerOrbMatchingName(entity!!.displayName.formattedText)
-            orb != null
+        ) {
+            it?.let { PowerOrbs.getPowerOrbMatchingName(it.displayName.formattedText) } != null
         }
         for (orbEntity in orbs) {
-            if (orbEntity == null) return
-            val orb = PowerOrbs.getPowerOrbMatchingName(orbEntity.displayName.formattedText)
-            if (orb != null && orb.ordinal >= heldOrb.ordinal) {
-                if (orbEntity.getDistanceSqToEntity(mc.thePlayer) <= orb.radius.pow(2.0)) {
-                    mc.thePlayer.playSound("random.orb", 0.8f, 1f)
-                    event.isCanceled = true
-                    break
-                }
-            }
+            orbEntity ?: continue
+            val orb = PowerOrbs.getPowerOrbMatchingName(orbEntity.displayName.formattedText) ?: continue
+            if (orb.ordinal < heldOrb.ordinal) continue
+            if (!orb.isPlayerInRange(orbEntity, mc.thePlayer)) continue
+            val remainingDuration = orb.getDurationRemaining(orbEntity)
+            if (remainingDuration != null && remainingDuration < Skytils.config.powerOrbDuration) continue
+            mc.thePlayer.playSound("random.orb", 0.8f, 1f)
+            event.isCanceled = true
+            break
         }
     }
 
     private enum class PowerOrbs(var orbName: String, var radius: Double, var itemId: String) {
-        RADIANT("§aRadiant", 18.0, "RADIANT_POWER_ORB"), MANAFLUX(
-            "§9Mana Flux",
-            18.0,
-            "MANA_FLUX_POWER_ORB"
-        ),
-        OVERFLUX("§5Overflux", 18.0, "OVERFLUX_POWER_ORB"), PLASMAFLUX("§d§lPlasmaflux", 20.0, "PLASMAFLUX_POWER_ORB");
+        RADIANT("§aRadiant", 18.0, "RADIANT_POWER_ORB"),
+        MANAFLUX("§9Mana Flux", 18.0, "MANA_FLUX_POWER_ORB"),
+        OVERFLUX("§5Overflux", 18.0, "OVERFLUX_POWER_ORB"),
+        PLASMAFLUX("§d§lPlasmaflux", 20.0, "PLASMAFLUX_POWER_ORB");
+
+        fun getDurationRemaining(entity: EntityArmorStand): Int? {
+            val durationText = entity.displayName.formattedText.substring(
+                orbName.length + 1,
```

Comment:
might break in the future but ok

Reply:
If it does, at least it shouldn't crash

Reply:
technically it could be out of bounds but ok